File: Observer/python-source/observers/ConsoleObserver.py
from __future__ import annotations

from .Observer import Observer


class ConsoleObserver(Observer):
    """Печатает каждое сообщение."""

    def __init__(self, name: str) -> None:
        self._name = name

    def update(self, message: str) -> None:
        print(f"{self._name} received: {message}")

File: JavaSource/src/main/java/ru/akhcheck/patterns/observer/ConsoleObserver.java
package ru.akhcheck.patterns.observer;

/** Печатает каждое сообщение. */
public class ConsoleObserver implements Observer {
    private final String name;

    public ConsoleObserver(String name) {
        this.name = name;
    }

    @Override
    public void update(String message) {
        System.out.println(name + " received: " + message);
    }
}
